### PR TITLE
refactor: add config struct and use slog

### PR DESCRIPTION
**Pull request by gligneul**
- Move node to internal package
- Add a struct that contains all node config
- Add structured logging with slog
- Add CARTESI_LOG_PRETTY
- Remove CARTESI_LOG_TIMESTAMP
- Log Rust service's messages in the correct level
- Print cartesi-rollups-cli to stdout

Co-Author: Victor Yves Crispim <rotcivyves@hotmail.com>
Co-Author: Francisco Moura <franciscoeduardomoura@gmail.com>

### CODE REVIEW

**Review by fmoura — DISMISSED** [review]

**Comment by endersonmaia** [comment]
While testing the logs, I see some `\` escaping and I don't think they're needed, right?

```
validator-1  | time=2024-03-15T11:08:45.413Z level=INFO msg="INFO dispatcher::machine::rollups_broker: producing input event event=RollupsInput { parent_id: \"0\", epoch_index: 0, inputs_sent_count: 1, data: AdvanceStateInput(RollupsAdvanceStateInput { metadata: InputMetadata { msg_sender: f39fd6e51aad88f6f4ce6ab8827279cfffb92266, block_number: 13, timestamp: 1710500920, epoch_index: 0, input_index: 0 }, payload: 3q2+7w==, tx_hash: 71932220268ade67936c208a119841cf9ef5b8b3b36ab380740fd814401baf2f }) }" service=dispatcher
```

This when sending an input:

```
validator-1  | time=2024-03-15T11:08:45.426Z level=INFO msg="\x1b[0m\x1b[38;5;8m[\x1b[0m\x1b[0m\x1b[32mINFO \x1b[0m rollup_http_server::http_service\x1b[0m\x1b[38;5;8m]\x1b[0m Received new request of type ADVANCE" service=server-manager
```

**Comment by gligneul** [comment]
@endersonmaia they are added automatically because they are already inside another string. `msg="INFO...`

**Comment by endersonmaia** [comment]
> @endersonmaia they are added automatically because they are already inside another string. `msg="INFO...`

Yeah, they're useless and are extra bytes in the log stream. Is it hard to avoid/remove them?

And what about the second example? 

`msg="\x1b[0m\x1b[38;5;8m[\x1b[0m\x1b[0m\x1b[32mINFO \x1b[0m` 

Looks like terminal coloring from the inner rust service logs.

**Review by torives — COMMENTED** [review]
Oh boy, this is looking neat! I loved how `text/template` made things so much easier to understand. I've only left a request concerning the logs and other small things.

**Comment by torives** [comment]
> > @endersonmaia they are added automatically because they are already inside another string. `msg="INFO...`
> 
> Yeah, they're useless and are extra bytes in the log stream. Is it hard to avoid/remove them?
> 
> And what about the second example?
> 
> `msg="\x1b[0m\x1b[38;5;8m[\x1b[0m\x1b[0m\x1b[32mINFO \x1b[0m`
> 
> Looks like terminal coloring from the inner rust service logs.

Can you share your environment to get such a message? I couldn't reproduce it by running it locally on my machine or with Docker. This wasn't happening on the original commit 🤔

File: cmd/cartesi-rollups-cli/root/deps/deps.go [hunk 1]
```diff
@@ -55,7 +55,7 @@ func run(cmd *cobra.Command, args []string) {
 	depsContainers, err := deps.Run(ctx, *depsConfig)
 	cobra.CheckErr(err)
 
-	config.InfoLogger.Println("all deps are up")
+	fmt.Println("all deps are up")
```
Comment (gligneul) [inline cmd/cartesi-rollups-cli/root/deps/deps.go:58]: use slog

File: cmd/cartesi-rollups-node/main.go [hunk 2]
```diff
@@ -5,73 +5,68 @@ package main
...
+			Level:     c.Log.Level,
+			AddSource: c.Log.Level == slog.LevelDebug,
+		}
+		return tint.NewHandler(os.Stdout, opts)
+	} else {
+		opts := &slog.HandlerOptions{
+			Level:     c.Log.Level,
+			AddSource: c.Log.Level == slog.LevelDebug,
+		}
+		return slog.NewTextHandler(os.Stdout, opts)
```
[the remark this answers is not in the document]
Comment (gligneul) [reply cmd/cartesi-rollups-node/main.go:70]: > You also didn't use the go-isatty lib. Was it intentional?

I will add it

**Review by torives — APPROVED** [review]
LGTM 🚀

**Review by fmoura — APPROVED** [review]

**Review by torives — APPROVED** [review]

**Pull request merged**